internal/consensus/state.go

```diff
@@ -1223,15 +1223,16 @@ func (cs *State) isProposer(address []byte) bool {
 func (cs *State) defaultDecideProposal(ctx context.Context, height int64, round int32) {
 	var block *types.Block
 	var blockParts *types.PartSet
+	var err error
 
 	// Decide on block
 	if cs.ValidBlock != nil {
 		// If there is valid block, choose that.
 		block, blockParts = cs.ValidBlock, cs.ValidBlockParts
 	} else {
 		// Create a new proposal block from state/txs from the mempool.
-		block, blockParts = cs.createProposalBlock()
-		if block == nil {
+		block, blockParts, err = cs.createProposalBlock()
```

Comment:
I believe this `err` isn't needed outside this scope.

```suggestion
		var err error
		block, blockParts, err = cs.createProposalBlock()
```